internal/credsretriever/refreshing_cache.go

```diff
@@ -138,17 +159,31 @@ func (r *cachedCredentialRetriever) GetIamCredentials(ctx context.Context,
 		return nil, nil, fmt.Errorf("service account is empty, cannot fetch credentials without a valid one")
 	}
 
-	// check if the request is in the cache, if it is, return it
-	if val, ok := r.internalCache.Get(request.ServiceAccountToken); ok {
-		if _, withinTtl := r.credentialsInEntryWithinValidTtl(val); withinTtl {
-			log.WithField("cache-hit", 1).Tracef("Using cached credentials")
-			return val.credentials, nil, nil
+	for i := 0; i <= defaultActiveRequestRetries; i++ {
+		// Check if the request is in the cache, if it is, return it
+		if val, ok := r.internalCache.Get(request.ServiceAccountToken); ok {
+			if _, withinTtl := r.credentialsInEntryWithinValidTtl(val); withinTtl {
+				log.WithField("cache-hit", 1).Tracef("Using cached credentials")
+				return val.credentials, nil, nil
+			}
+
+			log.Info("Identified that entry in cache contains credentials with small ttl or invalid ttl, will be deleted")
+			r.internalCache.Delete(request.ServiceAccountToken)
+			break
 		}
 
-		log.Info("Identified that entry in cache contains credentials with small ttl or invalid ttl, will be deleted")
-		r.internalCache.Delete(request.ServiceAccountToken)
+		if _, ok := r.internalActiveRequestCache.Get(request.ServiceAccountToken); ok {
+			// Wait 1s for active request to finish caching into internalCache
+			time.Sleep(1 * time.Second)
```

Comment:
Why hardcode to 1 second? Don't we want any backoff/jitter we ever retry more than once?

Reply:
See https://github.com/aws/eks-pod-identity-agent/pull/39#discussion_r1906997573 and https://github.com/aws/eks-pod-identity-agent/pull/39#discussion_r1907129175